tests/unit/l3-fprintf-perf-test.c

```diff
@@ -0,0 +1,92 @@
+/**
+ * *****************************************************************************
+ * \file l3-fprintf-perf-test.c
+ * \author Aditya P. Gurajada
+ * \brief L3: Lightweight Logging Library fprintf() logging APIs Unit-test
+ * \version 0.1
+ * \date 2024-05-29
+ *
+ * \copyright Copyright (c) 2024
+ *
+ * Usage: program-name [ <number-of-million-msgs> ]
+ *  Default, run fprintf()-logging perf-tests on 1 million messages
+ * *****************************************************************************
+ */
+#define _POSIX_C_SOURCE 199309L
+
+#include <stdlib.h>
+#include <inttypes.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "l3.h"
+#include "l3-perf-test.h"
+
+void test_fprintf_logging_perf(int nMil);
+
+/**
+ * *****************************************************************************
+ * The L3 logging-APIs to exercise logging using fprintf() system call are
+ * exercised in this unit-test. As a convenience, we just log a large # of
+ * entries to get a rough measure of logging throughput using this interface.
+ * *****************************************************************************
+ */
+int
+main(const int argc, const char * argv[])
+{
+    int nMil = 1;
+    if (argc > 1) {
+        nMil = atoi(argv[1]);
+    }
+
+    test_fprintf_logging_perf(nMil);
+
+    return 0;
+}
+
+#if L3_LOGT_FPRINTF
+
+void
+test_fprintf_logging_perf(int nMil)
```

Comment:
Define this conditionally, so that the build-time requirement of specifying `-DL3_LOGT_FPRINTF` is rigidly enforced.